js/views/transactionsVw.js

```diff
@@ -281,6 +285,118 @@ module.exports = baseVw.extend({
     tabWrapper.append(orderShort.render().el);
   },
 
+  clickDownloadCSV: function(e){
+    var targBtn = $(e.target);
+    targBtn.closest('.btn').addClass('loading');
+    this.downloadCSV(targBtn);
```

Comment:
As I've commented below, if you have the downloadCSV function return the Promise that's returned by the $.when call, then you wouldn't need to pass around the button. You could simply do:

```javascript
this.downloadCSV.always(() => {
    targBtn.removeClass('loading');
});
```